### PR TITLE
CORS-3195: Azure CAPI Supplemental Load Balancers

### PR DESCRIPTION
**Pull request by patrickdillon**
Adds support for creating an additional, external LB. As well as updating the capi-provisioned internal LB with config for the machine-config server.

/cc @rna-afk @jhixson74

### CODE REVIEW

File: pkg/infrastructure/azure/loadbalancer.go [hunk 1]
```diff
@@ -0,0 +1,238 @@
...
+	"path"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/to"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v4"
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/network/armnetwork/v2"
+)
+
+func createPublicIP(ctx context.Context, infraID, region string, pipClient *armnetwork.PublicIPAddressesClient) (*armnetwork.PublicIPAddress, error) {
+	publicIPAddressName := fmt.Sprintf("%s-pip-v4", infraID)
+	resourceGroupName := fmt.Sprintf("%s-rg", infraID)
```
Comment (patrickdillon) [inline pkg/infrastructure/azure/loadbalancer.go:16]: need to make all of these resourcegroupnames dynamic to account for user-provided values

File: pkg/asset/manifests/azure/cluster.go [hunk 2]
```diff
@@ -95,6 +95,15 @@ func GenerateClusterAssets(installConfig *installconfig.InstallConfig, clusterID
...
 				},
+				APIServerLB: capz.LoadBalancerSpec{
+					Name: fmt.Sprintf("%s-internal", clusterID.InfraID),
+					BackendPool: capz.BackendPool{
+						Name: fmt.Sprintf("%s-internal", clusterID.InfraID),
+					},
+					LoadBalancerClassSpec: capz.LoadBalancerClassSpec{
+						Type: capz.Internal,
+					},
+				},
```
Comment (jhixson74) [inline pkg/asset/manifests/azure/cluster.go:106]: This is duplicated.

**Comment by jhixson74** [comment]
There are some gofmt and golint errors. Other than that, everything LGTM.

**Comment by patrickdillon** [comment]
Adding a [ControlPlaneOutboundLB](https://capz.sigs.k8s.io/topics/control-plane-outbound-lb) to allow outbound access to the bootstrap node:
```
+                               ControlPlaneOutboundLB: &capz.LoadBalancerSpec{
+                                       FrontendIPsCount: to.Ptr(int32(1)),
+                               },

```
 and adding the bootstrap ignition commits, this enables control plane nodes to pull ignition:

**Comment by patrickdillon** [comment]
/retest

@jhixson74 I think the linting issues are fixed

**Comment by nrb** [comment]
Created https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/4755 to track implementing dual load balancer/endpoint logic upstream (a 4.17 concern), as well as creating [OCPCLOUD-2588](https://issues.redhat.com/browse/OCPCLOUD-2588)

**Pull request merged**